src/layout/Datepicker/index.tsx

```diff
@@ -40,6 +47,62 @@ export class Datepicker extends FormComponent<'Datepicker'> {
       />
     );
   }
+
+  runComponentValidation(node: LayoutNodeFromType<'Datepicker'>, overrideFormData?: IFormData): IValidationObject[] {
+    const state: IRuntimeState = window.reduxStore.getState();
+    const { langAsString, selectedLanguage } = staticUseLanguageFromState(state);
```

Comment:
I'd suggest making `langTools` and `formData` arguments passed into this function instead of using this 'cheat' to fetch the current state. Possibly make an object/type for the parameters passed into these functions so that you can pick out values you need from the object without relying on the parameter order.